targets/psa_switch/externs/psa_random.cpp

```diff
@@ -0,0 +1,74 @@
+/* Copyright 2020-present Cornell University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * Yunhe Liu (yunheliu@cs.cornell.edu)
+ *
+ */
+
+#include "psa_random.h"
+#include <bm/bm_sim/logger.h>
+
+#include <random>
+#include <thread>
+#include <iostream>
+
+namespace bm {
+
+namespace psa {
+
+void
+PSA_Random::init() {
+  valid_range = true;
+  min_val = min.get_uint64();
+  max_val = max.get_uint64();
+  range = max_val - min_val + 1;
+
+  if (min_val > max_val) {
+    valid_range = false;
+    Logger::get()->warn("Error: PSA extern random range must be positive.");
+  }
```

Comment:
Now that the check is in the `init` function, I would actually recommend using `_BM_ASSERT` so that the error is not missed. Then you can remove `valid_range` altogether.

We need a better way to report errors from `init`, but that can be done in the future.

Reply:
Done.